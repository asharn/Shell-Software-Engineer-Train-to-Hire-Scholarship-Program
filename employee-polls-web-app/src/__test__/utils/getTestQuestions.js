  let questions = {
    "test-user-id-one": {
      id: 'test-user-id-one',
      author: 'test-user-id-one',
      timestamp: 1467166872634,
      optionOne: {
        votes: ['test-user-id'],
        text: 'Build our new application with Javascript',
      },
      optionTwo: {
        votes: ['test-user-id-one'],
        text: 'Build our new application with Typescript'
      }
    },
    "test-user-id-two": {
      id: 'test-user-id-two',
      author: 'test-user-id',
      timestamp: 1468479767190,
      optionOne: {
        votes: [],
        text: 'hire more frontend developers',
      },
      optionTwo: {
        votes: ['test-user-id', 'test-user-id-one'],
        text: 'hire more backend developers'
      }
    },
    "test-user-id-three": {
      id: 'test-user-id-three',
      author: 'test-user-id-one',
      timestamp: 1488579767190,
      optionOne: {
        votes: [],
        text: 'conduct a release retrospective 1 week after a release',
      },
      optionTwo: {
        votes: ['test-user-id-one'],
        text: 'conduct release retrospectives quarterly'
      }
    },
    "test-user-id-four": {
      id: 'test-user-id-four',
      author: 'test-user-id',
      timestamp: 1482579767190,
      optionOne: {
        votes: [],
        text: 'have code reviews conducted by peers',
      },
      optionTwo: {
        votes: ['test-user-id-one'],
        text: 'have code reviews conducted by managers'
      }
    }
  }

  export function getTestQuestions () {
    return new Promise((resolve) => {
      setTimeout(() => resolve({...questions}), 1000)
    })
  }