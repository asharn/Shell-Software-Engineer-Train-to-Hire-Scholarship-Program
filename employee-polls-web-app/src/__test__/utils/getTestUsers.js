const users = {
    'test-user-id': {
      id: 'test-user-id',
      password:'test-pass',
      name: 'Test User',
      avatarURL: '/avatars/test-user-id.jpg',
      answers: {
        "test-question-id-one": 'optionOne'
      },
      questions: ['test-question-id-two']
    },
    'test-user-id-one': {
        id: 'test-user-id-one',
        password:'test-pass',
        name: 'Test User',
        avatarURL: '/avatars/test-user-id.jpg',
        answers: {
          "test-question-id-two": 'optionOne'
        },
        questions: ['test-question-id-three']
      }
  }

  export function getTestUsers () {
    return new Promise((resolve) => {
      setTimeout(() => resolve({...users}), 1000)
    })
  }