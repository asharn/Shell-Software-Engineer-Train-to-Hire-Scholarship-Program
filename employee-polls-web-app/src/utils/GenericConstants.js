//UI ids
export const OPTION_ONE = 'optionOne'; 
export const OPTION_TWO = 'optionTwo'; 

//NavBar related constans
export const HOME = 'Home';
export const DASHBOARD = 'Dashboard';
export const LEADERBOARD = 'Leaderboard';
export const NEW = 'New';
export const SIGN_IN = 'SignIn';
export const SIGN_OUT = 'SignOut';
export const SIGN_UP = 'SignUp';

//Company related constants
export const COMPANY_NAME = 'Employee Polls';

//Text shown in buttons
export const SHOW = 'Show';
export const SUBMIT = 'Submit';
export const BUTTON_TEXT_SIGN_UP = 'Sign Up';
export const BUTTON_TEXT_SIGN_IN = 'Sign In';

export const BUTTON_TEXT_CLICK_OPTION_ONE = '- Click Option One -';
export const BUTTON_TEXT_CLICK_OPTION_TWO = '- Click Option Two -';
export const BUTTON_TEXT_VOTES = 'Votes';



//Dashboard related constants
export const NEW_QUESTION = 'New Question';
export const ANSWERED_QUESTION = 'Answered Question';



//Genric texts
export const CREATE_NEW_POLLS = 'Create New Poll:';
export const WOULD_YOU_RATHER = 'Would you rather...';
export const YOU_ARE_NOT_AUTHORIZED = 'You are not authorized to visit this page.';
export const NO_MATCH_FOR = 'No match for ';
export const POLL_BY = 'Poll By : ';
export const FORGOT_PASSWORD_WHAT = 'Forgot password?';
export const DONT_HAVE_AN_ACCOUNT = 'Don\'t have an account? Sign Up';
export const ALREADY_HAVE_AN_ACCOUNT = 'Already have an account? Sign in';
export const COMMING_SOON = 'Comming Soon...';










