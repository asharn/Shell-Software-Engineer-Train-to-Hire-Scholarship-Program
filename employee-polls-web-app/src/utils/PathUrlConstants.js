export const RootPathUrl = '/'; 
export const DashBoardUrl = '/dashboard'; 
export const LeaderBoardUrl = '/leaderboard'; 
export const SignInUrl = '/sign-in'; 
export const SignOutUrl = '/sign-out'; 
export const SignUpUrl = '/sign-up'; 
export const NewPollUrl = '/add'; 
export const PathParamId = '/:id';
export const QuestionUrl = '/question'; 
export const QuestionWithIdUrl = QuestionUrl+PathParamId; 
export const OtherThenMapUrl = '/*'; 
export const ForgotPasswordUrl = '/forgot-password'; 



