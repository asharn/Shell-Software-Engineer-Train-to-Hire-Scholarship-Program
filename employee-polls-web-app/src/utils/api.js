import {
  _getUsers,
  _getQuestions,
  _saveQuestion,
  _saveQuestionAnswer,
} from './_DATA.js'

export function getInitialData () {
  return Promise.all([
    _getUsers(),
    _getQuestions(),
  ]).then(([users, questions]) => ({
    users,
    questions,
  }))
}

export function saveQuestion (question) {
  return _saveQuestion(question)
}

export function saveQuestionAnswer ( authedUser, qid, answer ) {
  console.log({authedUser, qid, answer });
  return _saveQuestionAnswer( authedUser, qid, answer )
}