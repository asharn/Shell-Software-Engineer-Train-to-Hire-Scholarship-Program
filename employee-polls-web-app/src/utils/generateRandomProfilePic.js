function randomInteger(max) {
    return Math.floor(Math.random()*(max + 1));
}

function randomRgbColor() {
    let r = randomInteger(255);
    let g = randomInteger(255);
    let b = randomInteger(255);
    return [r,g,b];
}

function randomHexColor() {
    let [r,g,b] =randomRgbColor();
 
    let hr = r.toString(16).padStart(2, '0');
    let hg = g.toString(16).padStart(2, '0');
    let hb = b.toString(16).padStart(2, '0');
 
    return hr + hg + hb;
}

function getListOfColor(randomeNumber) {
    let colors = [];
    for(let i = 0; i <= randomeNumber; i++) {
        colors.push(randomHexColor());
    }
 
    return colors;
}


export function generateRandomProfilePic (username) {

    const listOfVariant = ['marble', 'beam', 'pixel', 'sunset', 'ring' ,  'bauhaus'];
    let variant = listOfVariant[Math.floor(Math.random()*listOfVariant.length)];
    let colors = getListOfColor(randomInteger(10));
    console.log(colors);
    let avatarUrl = 'https://source.boringavatars.com/' + variant+'/192/'+username+'?colors='+colors.join(',');
    return avatarUrl;
  }