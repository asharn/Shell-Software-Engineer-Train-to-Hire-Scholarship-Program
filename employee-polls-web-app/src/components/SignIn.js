import * as React from 'react';
import Avatar from '@mui/material/Avatar';
import Button from '@mui/material/Button';
import CssBaseline from '@mui/material/CssBaseline';
import TextField from '@mui/material/TextField';
import FormControlLabel from '@mui/material/FormControlLabel';
import Checkbox from '@mui/material/Checkbox';
import Link from '@mui/material/Link';
import Grid from '@mui/material/Grid';
import Box from '@mui/material/Box';
import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
import Visibility from '@mui/icons-material/Visibility';
import VisibilityOff from '@mui/icons-material/VisibilityOff';
import Typography from '@mui/material/Typography';
import Container from '@mui/material/Container';
import { createTheme, ThemeProvider } from '@mui/material/styles';
import { setAuthedUser } from '../actions/authedUser';
import { connect } from 'react-redux';
import { IconButton, InputAdornment } from '@mui/material';
import { useNavigate, useLocation, useParams} from "react-router-dom";
import { ForgotPasswordUrl, RootPathUrl, SignUpUrl} from '../utils/PathUrlConstants';
import { BUTTON_TEXT_SIGN_IN,FORGOT_PASSWORD_WHAT,DONT_HAVE_AN_ACCOUNT } from '../utils/GenericConstants';


const withRouter = (Component) => {
  const ComponentWithRouterProp = (props) => {
    let location = useLocation();
    let navigate = useNavigate();
    let params = useParams();
    return <Component {...props} router={{ location, navigate, params }} />;
  };

  return ComponentWithRouterProp;
};

const theme = createTheme();

const SignIn = (props) => {
  const navigate = useNavigate();
  const search = useLocation().search;
  const [showPassword, setShowPassword] = React.useState(false);
  const [success, setSuccess] = React.useState(false);
  const [error, setError] = React.useState(false);
  const [usernameValue, setUsernameValue] = React.useState('');
  const [passwordValue, setPasswordValue] = React.useState('');

  const handleClickShowPassword = () => setShowPassword(!showPassword);
  const { users} = props;

  const handleUsernameChange = (e) => {
      setUsernameValue(e.target.value);
      setError(false);
  };

  const handlePasswordChange = (e) => {
      setPasswordValue(e.target.value);
      setError(false);
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    const data = new FormData(event.currentTarget);
    const username = data.get('username');
    const password = data.get('password');
    
    if(username === ''){
      setSuccess(false);
      setError(true);
    }else if(password === ''){
      setSuccess(false);
      setError(true);
    }else{
      if(users[username] && users[username].password === password){
        new Promise((res, rej) => {
          setTimeout(() => res(), 500);
        }).then(() => {
          props.dispatch(setAuthedUser(username));
          setSuccess(true);
          setError(false);
          const redirectTo = new URLSearchParams(search).get('redirectTo');
          redirectTo && redirectTo!=='' ? navigate(redirectTo) : navigate(RootPathUrl);
        });
      }else{
        setSuccess(false);
        setError(true);
      }

    }    
  };

  return (
    <ThemeProvider theme={theme}>
      <Container component="main" maxWidth="xs">
        <CssBaseline />
        <Box
          sx={{
            marginTop: 8,
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <Avatar sx={{ m: 1, bgcolor: 'secondary.main' }}>
            <LockOutlinedIcon />
          </Avatar>
          <Typography component="h1" variant="h5">
            Sign in
          </Typography>
          {success ?
              <Typography color='success' className={"Success"} data-testid="success-header">Login Successfully!</Typography>
            :
              <Typography color='error' className={"Error"} data-testid="error-header">&nbsp;{error && 'Username or Password is invalid.'}&nbsp;</Typography>
            }
          <Box component="form" onSubmit={handleSubmit} sx={{ mt: 1 }}>
            <TextField
              margin="normal"
              required
              fullWidth
              id="username"
              data-testid="username"
              label="Username"
              name="username"
              autoComplete="username"
              value={usernameValue}
              onChange={handleUsernameChange}
              autoFocus
            />
            <TextField
              margin="normal"
              required
              fullWidth
              name="password"
              label="Password"
              value={passwordValue}
              onChange={handlePasswordChange}
              type={showPassword ? "text" : "password"}
              id="password"
              data-testid="password"
              autoComplete="current-password"
              InputProps={{ 
                endAdornment: (
                  <InputAdornment position="end">
                    <IconButton
                      aria-label="toggle password visibility"
                      onClick={handleClickShowPassword}
                    >
                      {showPassword ? <Visibility /> : <VisibilityOff />}
                    </IconButton>
                  </InputAdornment>
                )
              }}
            />
            <FormControlLabel
              control={<Checkbox value="remember" color="primary" />}
              label="Remember me"
            />
            <Button
              type="submit"
              fullWidth
              variant="contained"
              sx={{ mt: 3, mb: 2 }}
              data-testid='sign-in'
            >
              {BUTTON_TEXT_SIGN_IN}
            </Button>
            <Grid container>
              <Grid item xs>
                <Link href={ForgotPasswordUrl+search} variant="body2">
                  {FORGOT_PASSWORD_WHAT}
                </Link>
              </Grid>
              <Grid item>
                <Link href={SignUpUrl+search} variant="body2">
                  {DONT_HAVE_AN_ACCOUNT}
                </Link>
              </Grid>
            </Grid>
          </Box>
        </Box>
      </Container>
    </ThemeProvider>
  );
}

function mapStateToProps({ authedUser, users }, props) {
  return {
    authedUser,
    users
  };
}

export default withRouter(connect(mapStateToProps)(SignIn));