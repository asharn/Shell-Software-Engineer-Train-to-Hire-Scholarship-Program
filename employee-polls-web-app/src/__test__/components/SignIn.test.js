import { render, screen } from '@testing-library/react';
import SignIn from '../../components/SignIn';
import configureStore from 'redux-mock-store';
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { Router } from 'react-router';
import {createMemoryHistory} from 'history';
import {getTestUsers} from '../utils/getTestUsers'


const mockStore = configureStore([]);


describe('Snapshot testing for SignIn component.', () => {
  
  let store;
    let component;

    let users = getTestUsers();
    const history = createMemoryHistory();
    beforeEach(() => {
      store = mockStore({
        authedUser: 'test-user-id',
        users: users
      });
      
      component = renderer.create(
        <Provider store={store}>
          <Router location={history.location} navigator={history}><SignIn /></Router>
        </Provider>
      );
    });
    
  test('renders SignIn component with screen.debug().', async () => {
    
    render(<Provider store={store}>
      <Router location={history.location} navigator={history}><SignIn /></Router>
    </Provider>);
    screen.debug();
  });

  test('renders SignIn component with toMatchSnapshot.', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

});

describe('Verify username, password and submit button is present.', () => {
  
  let store;
    let component;

    let users = getTestUsers();
    const history = createMemoryHistory();
    beforeEach(() => {
      store = mockStore({
        authedUser: 'test-user-id',
        users: users
      });
      
      component = renderer.create(
        <Provider store={store}>
          <Router location={history.location} navigator={history}><SignIn /></Router>
        </Provider>
      );
    });
    
  test('Veriy username field is there in the SignIn component.', () => {
    component = render(<Provider store={store}>
                  <Router location={history.location} navigator={history}><SignIn /></Router>
                </Provider>);
        var username = component.getByTestId('username')
        expect(username).toBeInTheDocument();
    
  });

  test('Veriy password field is there in the SignIn component.', () => {
    
    component = render(<Provider store={store}>
          <Router location={history.location} navigator={history}><SignIn /></Router>
        </Provider>);
        var password = component.getByTestId('password')
        expect(password).toBeInTheDocument();
    
  });

  test('Veriy Sign In button is there in the SignIn component.', () => {
    component = render(<Provider store={store}>
          <Router location={history.location} navigator={history}><SignIn /></Router>
        </Provider>);
        var signInButton = component.getByTestId('sign-in')
        expect(signInButton).toBeInTheDocument();
    
  });

});