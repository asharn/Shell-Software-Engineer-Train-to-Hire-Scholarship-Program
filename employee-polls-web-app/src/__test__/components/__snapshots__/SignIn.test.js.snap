// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing for SignIn component. renders SignIn component with toMatchSnapshot. 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    className="MuiBox-root css-binzgt"
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1999axt-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign in
    </h1>
    <p
      className="MuiTypography-root MuiTypography-body1 Error css-1nkhxq8-MuiTypography-root"
      data-testid="error-header"
    >
       
       
    </p>
    <form
      className="MuiBox-root css-164r41r"
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        data-testid="username"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="username"
          id="username-label"
        >
          Username
          <span
            aria-hidden={true}
            className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="username"
            autoFocus={true}
            className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="username"
            name="username"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Username
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
        data-testid="password"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="password"
          id="password-label"
        >
          Password
          <span
            aria-hidden={true}
            className="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
          >
             
            *
          </span>
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            autoComplete="current-password"
            autoFocus={false}
            className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
            disabled={false}
            id="password"
            name="password"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={true}
            type="password"
            value=""
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          >
            <button
              aria-label="toggle password visibility"
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VisibilityOffIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-1ftyaf0"
            >
              <span>
                Password
                 
                *
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <label
        className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
      >
        <span
          className="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root"
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={null}
        >
          <input
            className="PrivateSwitchBase-input css-1m9pwf3"
            data-indeterminate={false}
            onChange={[Function]}
            type="checkbox"
            value="remember"
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CheckBoxOutlineBlankIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
            />
          </svg>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </span>
        <span
          className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-ahj2mt-MuiTypography-root"
        >
          Remember me
        </span>
      </label>
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth MuiButtonBase-root  css-1vhaqj4-MuiButtonBase-root-MuiButton-root"
        data-testid="sign-in"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Sign In
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <a
            className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
            href="/forgot-password"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Forgot password?
          </a>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <a
            className="MuiTypography-root MuiTypography-body2 MuiLink-root MuiLink-underlineAlways css-101ca9i-MuiTypography-root-MuiLink-root"
            href="/sign-up"
            onBlur={[Function]}
            onFocus={[Function]}
          >
            Don't have an account? Sign Up
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
